### PR TITLE
#1458: fix report counting for general/execution categories

### LINKED ISSUE #1458: Number of passing products in summary does not match actual total
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When I validate products in a directory, the summary at the end of the report sometimes shows more products passed than there are total products.

### 🕵️ Expected behavior

I expected the number of passing products to be no more than the total number of products.

### 📜 To Reproduce

Using the attached test data, run validate:

```bash
validate --skip-content-validation -t spurious_pass
```

Observe that there is one more product that than the total number of products:

```
  Product Validation Summary:
    3          product(s) passed
    0          product(s) failed
    0          product(s) skipped
    2          product(s) total
```

### 🖥 Environment Info

Ubuntu Linux 22.04
Java 17

### 📚 Version of Software Used

validate-4.0.8

### 🩺 Test Data / Additional context

[spurious_pass.zip](https://github.com/user-attachments/files/24201059/spurious_pass.zip)

[spurious_pass.txt](https://github.com/user-attachments/files/24201060/spurious_pass.txt)

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Do a better job of ignoring the general and execution categories

### 🤖 AI Assistance Disclosure
- [x] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## ♻️ Related Issues
Closes #1458 

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation

All automated unit/regression tests should pass. ~~Due to lack of sample data, could not implement test for this specific case. Waiting response from reporter for missing file.~~

- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

**Review by nutjob4life — APPROVED** [review]
Code delta: ✓
Maven tests: ✓
Approval: ✅
Maven details:
```
292 scenarios (257 passed, 35 failed)
876 steps (841 passed, 35 failed)
11m 35.27s
Mar 23, 2026 3:46:19 PM org.junit.platform.launcher.core.DiscoveryIssueNotifier logIssues
WARNING: TestEngine with ID 'cucumber' encountered a non-critical issue during test discovery:

(1) [WARNING] Discovering tests using the cucumber.features property. Other discovery selectors are ignored!

This is a work around for the limited JUnit 5 support in Maven and Gradle. Please request/upvote/sponsor/ect better support for JUnit 5 discovery selectors. For details see: https://github.com/cucumber/cucumber-jvm/pull/2498

If you are using the JUnit 5 Suite Engine, Platform Launcher API or Console Launcher you should not use this property. Please consult the JUnit 5 documentation on test selection.
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 695.1 s -- in io.cucumber.junit.platform.engine.CucumberTestEngine
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 23
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

File: src/test/resources/github1458/obj.xml [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+        <Investigation_Area>
+            <name>OSIRIS-REx</name>
+            <type>Mission</type>
+            <Internal_Reference>
+                <lid_reference>urn:nasa:pds:context:investigation:mission.orex</lid_reference>
+                <reference_type>data_to_investigation</reference_type>
+            </Internal_Reference>
+        </Investigation_Area>
+        <Observing_System>
+            <name>Origins, Spectral Interpreation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx)</name>
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/test/resources/github1458/obj.xml:53]: ignoring. this is test data.

File: src/test/resources/github1458/obj.xml [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+                    <data_type>ASCII_Real</data_type>
+                    <unit>m/s</unit>
+                    <description>y-component of velocity at impact ('IAU_BENNU' SPICE frame)</description>
+                </Field_Delimited>
+                <Field_Delimited>
+                    <name>imapct_vz</name>
+                    <field_number>79</field_number>
+                    <data_type>ASCII_Real</data_type>
+                    <unit>m/s</unit>
+                    <description>z-component of velocity at impact ('IAU_BENNU' SPICE frame)</description>
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/test/resources/github1458/obj.xml:673]: ignoring. this is test data. it doesn't matter.

File: src/test/resources/github1458/orba_mp4.xml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            Lauretta, D. S.</author_list>
+      <editor_list>Crombie, M.K.</editor_list>
+      <publication_year>2025</publication_year>
+      <description> Derived Radio Science Cosmographia animation of Bennu particles from Jan. 6 - Feb. 20, 2019 (during Orbit A)</description>
+    </Citation_Information>
+    <Modification_History>
+      <Modification_Detail>
+        <modification_date>2025-11-25</modification_date>
+        <version_id>1.0</version_id>
+        <description>Inital label for this data product.</description>
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/test/resources/github1458/orba_mp4.xml:25]: ignoring. this is test data. it doesn't matter.

File: src/test/resources/features/4.1.x.feature [hunk 4]
```diff
@@ -6,4 +6,5 @@ Feature: 4.1.x
     @4.1.x
     Examples: 
       | issueNumber | subtest | datasrc | args | expectation |
-| 956 | 1 | "github956" | "--skip-context-validation -t {datasrc}/" | |
\ No newline at end of file
+| 956 | 1 | "github956" | "--skip-context-validation -t {datasrc}/" | |
+| 1458 | 1 | "github1458" | " -t {datasrc}/" | "summary:productValidation:passed=2,summary:productValidation:total=2,summary:totalWarnings=6,messageTypes:warning.label.context_ref_mismatch=5,messageTypes:warning.label.schematron=1" |
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/test/resources/features/4.1.x.feature:10]: tests pass. let's keep it as is for now.

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/test/resources/features/4.1.x.feature:10]: tests pass. let's keep it as is for now.

File: src/main/java/gov/nasa/pds/validate/report/Report.java [hunk 5]
```diff
@@ -170,15 +170,18 @@ final public Status record(URI sourceUri, final ValidationProblem problem) {
...
     // TODO: Handle null problems
-    int ignoreFromProductCounts = 0;
     for (ValidationProblem problem : problems) {
-      if (problem.getProblem().getSeverity() == ExceptionType.ERROR
+      ProblemCategory category = problem.getProblem().getType().getProblemCategory();
+      if (category.equals(ProblemCategory.GENERAL) || category.equals(ProblemCategory.EXECUTION)) {
+        ignoreFromTotalCounts++;
+      } else if (problem.getProblem().getSeverity() == ExceptionType.ERROR
           || problem.getProblem().getSeverity() == ExceptionType.FATAL) {
         if (ExceptionType.ERROR.getValue() <= this.level.getValue()) {
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/validate/report/Report.java:186]: Because of the way we extended logging, we need to capture these separately for counting purposes. this is the point of this issue. skipping this for now.
